src/client/orderbook/OrderbookFullClient.ts

```diff
@@ -519,4 +570,27 @@ export class OrderbookFullClient extends OrderbookReadClient {
       effects,
     };
   }
+
+  public async createSafeAndBidWithCommission(p: {
+    beneficiary: SuiAddress;
+    collection: string;
+    commission: number;
+    ft: string;
+    orderbook: ObjectId;
+    price: number;
+    wallet: ObjectId;
+  }) {
+    const effects = await this.client.sendTxWaitForEffects(
+      createSafeAndBidWithCommissionTx({
+        ...this.opts,
+        ...p,
+      })
+    );
+
+    return {
+      // undefined if trade not executed instantly
+      trade: effects.created?.find(Boolean)?.reference.objectId,
```

Comment:
again, this will end up being the `safe`, either remove it or add tests to return the correct object